db/migrate/20210224160921_remove_disabled_products_from_sync_plans.rb

```diff
@@ -0,0 +1,14 @@
+class RemoveDisabledProductsFromSyncPlans < ActiveRecord::Migration[6.0]
+  def change
+    ::Katello::SyncPlan.all.each do |sync_plan|
+      product_ids = []
+      sync_plan.products.each do |product|
+        product_ids << product.id if (product&.redhat? && product&.repositories&.count == 0)
+      end
+      if product_ids.size > 0
+        sync_plan.product_ids = sync_plan.product_ids - product_ids
+        sync_plan.save!
+      end
```

Comment:
I think you could do something simpler (and with better performance):
```
disabled = Katello::Product.redhat.where.not(:id => Katello::Product.enabled)
Katello::SyncPlanProduct.where(:product_id => disabled).delete_all
```

Reply:
ohhh there's no SyncPlanProduct model

Reply:
here we go:
```
disabled = Katello::Product.redhat.where.not(:id => Katello::Product.enabled)
disabled.update_all(:sync_plan_id => nil)
```

Note: untested :)

Reply:
This will fetch a lot of products for a big manifest..You think it's better to operate with sync_plans since there'll be fewer of those?

Reply:
it should execute as a single query which would be much more efficient

Reply:
Ok..I'll update and test.. :+1:

Reply:
Pushed a commit with the suggested changes..